src/Uno.UI/DataBinding/BindingExpression.cs

```diff
@@ -244,6 +249,11 @@ public void SetSourceValue(object value)
 
 			try
 			{
+				if (FrameworkTemplatePool.IsRecycling && _view.IsAlive && _view.Target is IFrameworkTemplatePoolAware)
```

Comment:
@MartinZikmund We should be adding a small explanation on why we're doing this.

Reply:
Will do